docs/concepts/errors.md

```diff
@@ -1,3 +1,60 @@
 # Errors
+Fyle uses conventional HTTP response codes to indicate the success or failure of an API request. 
+In general: 
+*   Codes in the 2xx range indicate success. 
+*   Codes in the 4xx range indicate an error that failed given the information provided (e.g., a required parameter is missing, improper usage of parameters etc.). 
+*   Codes in the 5xx range indicate an error with Fyle's servers(these are rare). 
 
-This section is yet to be written
+> Please contact us at support@fylehq.com if a 5xx error is encountered. 
+
+## Types Of Errors
+| SL NO.  |  HTTP STATUS CODE  | HTTP MESSAGE | ENGLISH EXPALINATION |
+|---------|--------------------|--------------|--------------------- |
+|    1.   |         400       |   Bad Request |  The request data or query sent is malformed |
+|     2.   |         401      |  Unauthorized |  The token is either not given Fyle or is timed out |
+|    3.   |          403      |     Forbidden |  The API token doesn't have permissions to perform the requested action|
+|     4.    |          404     |      Not Found  |  We dont understand request
+
+## Differnet Reasons for 400 error 
+
+1. **ValidationError:** 
+    This type error occurs when either a parameter with is mandatory is missing or when the parameters that are not formated as defined (Eg: Passing a interger for a string , Not following pattern required for phone numbers etc.)
+    <!--focus: false-->
+    |![Required Parameter Missig Error](../../assets/images/concepts/errors/ValidationError2.png)|
+    |:--:|
+    | <b>Example of ValidationError. Required parameter user_full_name is missing</b>    
+2. **BulkError:**
+    This type error occurs when you are doing a bulk operations. Generally this error occurs when you pass a value for an entity parameter for which that particular entity is not found. (Eg: Passing an unkonwn department name which does not exists etc.). 
+     <!--focus: false-->
+    |![Required Parameter Missig Error](../../assets/images/concepts/errors/bulkError.png)|
+    |:--:|
+    | <b>Example of BulkError. Department named XYZ does not exists in the org</b> 
+3. **IntegrityError:**
+    This error is encountered when an sql contraints fails. (Eg: Creating an project with same project name , sub project name inside the same org, passing user_id for which the users dont exists).
```

Comment:
Don't mention this as SQL constraint.. we should be cleaning up these and not passing them back as is. External developer can't access our DB

Reply:
I will just remove explanation for these kinds of error